packages/react-flex/src/components/Flex/Flex.tsx

```diff
@@ -1,5 +1,80 @@
 import * as React from 'react';
+import { FlexProps, FlexSlots, FlexSlotProps } from './Flex.types';
+import { compose, createClassResolver, mergeProps } from '@fluentui/react-compose';
+import * as classes from './Flex.scss';
+import { FlexItem } from '../FlexItem/FlexItem';
+import { FlexContext } from './FlexContext';
+import { useFlex } from './useFlex';
 
-export const Flex = () => {
-  return <div />;
+export const Flex = compose<'div', FlexProps, FlexProps, {}, {}>(
+  (props, ref, options) => {
+    const { children, disableShrink } = props;
+    const { state } = options;
+    const { slots, slotProps } = mergeProps<FlexProps, FlexProps, FlexSlots, FlexSlotProps>(state, options);
+    const gap: string = props.tokens && props.tokens.gap ? props.tokens.gap : '0';
+
+    const { flexChildren, generalChildren } = separateFlexChildren(children);
+
+    return (
+      <slots.root {...slotProps.root}>
+        <FlexContext.Provider value={{ disableShrink: !!disableShrink, gap: gap }}>{flexChildren}</FlexContext.Provider>
+        {generalChildren}
+      </slots.root>
+    );
+  },
+  {
+    classes: createClassResolver(classes),
+    displayName: 'Flex',
+    handledProps: [
+      'inline',
+      'column',
+      'wrap',
+      'horizontalAlign',
+      'verticalAlign',
+      'reverse',
+      'disableShrink',
+      'fluid',
+      'tokens',
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    ] as any,
+    state: useFlex,
+  },
+);
+
+const separateFlexChildren = (children: React.ReactNode) => {
+  const flexChildren: React.ReactNode[] = [];
+  const generalChildren: React.ReactNode[] = [];
+
+  React.Children.forEach(children, child => {
+    if (isFlexItemElement(child)) {
+      flexChildren.push(child);
+    } else {
+      generalChildren.push(child);
+    }
+  });
+
+  return { flexChildren, generalChildren };
+};
+
+const isFlexItemElement = (item: React.ReactNode): item is typeof FlexItem => {
+  return (
+    !!item &&
+    typeof item === 'object' &&
+    !!(item as React.ReactElement).type &&
+    ((item as React.ReactElement).type as React.ComponentType).displayName === FlexItem.displayName
+  );
+};
+
+Flex.defaultProps = {
+  as: 'div',
+  inline: false,
+  wrap: false,
+  horizontalAlign: 'start',
+  verticalAlign: 'stretch',
+  disableShrink: false,
+  fluid: false,
+  tokens: {
+    padding: '0',
+    gap: '0',
+  },
```

Comment:
We needed this for the `scss` styles `_true` and `_false` statements, right?

Reply:
yes, otherwise nested Flex components get parent's values.